meson.build

```diff
@@ -9,20 +9,23 @@ fwupd_version = meson.project_version()
 varr = fwupd_version.split('.')
 fwupd_major_version = varr[0]
 fwupd_minor_version = varr[1]
-fwupd_micro_version = varr[2].split('-')[0]
-conf = configuration_data()
+fwupd_micro_version = varr[2]
 
-if varr[2].contains('-')
-  fwupd_dirty_version = varr[2].split('-')[1]
-  fwupd_commit = varr[2].split('-')[2]
-  conf.set('FWUPD_DIRTY_VERSION', fwupd_dirty_version)
-  conf.set_quoted('FWUPD_COMMIT_VERSION', fwupd_commit)
-endif
+conf = configuration_data()
 conf.set('FWUPD_MAJOR_VERSION', fwupd_major_version)
 conf.set('FWUPD_MINOR_VERSION', fwupd_minor_version)
 conf.set('FWUPD_MICRO_VERSION', fwupd_micro_version)
 conf.set_quoted('PACKAGE_VERSION', fwupd_version)
 
+# non tagged builds
+git_version = run_command('git', 'describe').stdout().strip()
```

Comment:
to avoid making a hard git dependency I think you need to check for if git is available to run

Reply:
Good catch, fixed.